src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+};
+static T_DDL_AUDIT_HANDLE *ddl_audit_handle = NULL;
+
+static int cub_make_ddl_log_filename (char *filename_buf, size_t buf_size, T_APP_NAME app_name);
+static int cub_make_schema_file_name (const char *file_full_path, char *dest_path, size_t buf_size);
+static void cub_ddl_log_backup (const char *path);
+#if defined(WINDOWS)
+static void unix_style_path (char *path);
+#endif /* WINDOWS */
+static int cub_create_dir_log (const char *new_dir);
+static int cub_create_log_mgs (char *msg);
+static int cub_get_current_time (char *buf, size_t size);
+static int cub_file_copy (char *src_file, char *dest_file);
+static void cub_remove_char (char *string, char ch);
+static FILE *cub_ddl_log_open (T_APP_NAME app_name);
+static int cub_get_time_string (char *buf, struct timeval *time_val);
+static FILE *cub_fopen_and_lock (const char *path, const char *mode);
+static void cub_ddl_log_elapsed_time (long sec, long msec);
+static void cub_timeval_diff (struct timeval *start, struct timeval *end, int *res_sec, int *res_msec);
+static const char *cub_get_app_name (T_APP_NAME app_name);
+
+static char is_executed_ddl = FALSE;
+
+void
+cub_ddl_log_init ()
+{
+  if (ddl_audit_handle != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+
+  ddl_audit_handle = (T_DDL_AUDIT_HANDLE *) MALLOC (sizeof (T_DDL_AUDIT_HANDLE));
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  memset (ddl_audit_handle, 0x00, sizeof (T_DDL_AUDIT_HANDLE));
+
+  ddl_audit_handle->stmt_type = -1;
+  ddl_audit_handle->execute_type = DDL_LOG_RUN_EXECUTE_FUNC;
+  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
+}
+
+void
+cub_ddl_log_free (char all_free)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      cub_ddl_log_init ();
+      return;
+    }
+
+  FREE_MEM (ddl_audit_handle->sql_text);
+  FREE_MEM (ddl_audit_handle->err_msg);
+
+  ddl_audit_handle->stmt_type = -1;
+  ddl_audit_handle->execute_type = DDL_LOG_RUN_EXECUTE_FUNC;
+  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
+  ddl_audit_handle->log_filepath[0] = '\0';
+  ddl_audit_handle->execute_start_time[0] = '\0';
+  ddl_audit_handle->elapsed_time[0] = '\0';
+  ddl_audit_handle->file_line_number = 0;
+  ddl_audit_handle->err_code = 0;
+  ddl_audit_handle->msg[0] = '\0';
+  ddl_audit_handle->file_name[0] = '\0';
+  ddl_audit_handle->schema_file[0] = '\0';
+  ddl_audit_handle->commit_count = 0;
+
+  if (all_free)
+    {
+      ddl_audit_handle->auto_commit_mode = FALSE;
+      is_executed_ddl = FALSE;
+    }
+}
+
+void
+cub_ddl_log_destroy ()
+{
+  if (ddl_audit_handle != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+}
+
+void
+cub_ddl_log_app_name (T_APP_NAME app_name)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+  ddl_audit_handle->app_name = app_name;
+}
```

Comment:
if you just execute a single, or couple of statements, what's about this style:
```
void
cub_ddl_log_app_name (T_APP_NAME app_name)
{
   if (ddl_audit_handle)
     {
       ddl_audit_handle->app_name = app_name;
     }
 }
```

Reply:
Thank you, i will change code as your recommend.